bld_server/src/endpoints/hist.rs

```diff
@@ -1,28 +1,38 @@
 use crate::extractors::User;
-use actix_web::{get, web::Data, HttpResponse, Responder};
+use actix_web::{
+    get,
+    web::{Data, Query},
+    HttpResponse, Responder,
+};
 use bld_core::database::pipeline_runs;
 use diesel::r2d2::{ConnectionManager, Pool};
 use diesel::sqlite::SqliteConnection;
 use tracing::info;
 
-#[get("/hist")]
+#[get("/hist/state/:")]
 pub async fn hist(
+    query: Query<String>,
     user: Option<User>,
     db_pool: Data<Pool<ConnectionManager<SqliteConnection>>>,
 ) -> impl Responder {
     info!("Reached handler for /hist route");
     if user.is_none() {
         return HttpResponse::Unauthorized().body("");
     }
-    match history_info(db_pool.get_ref()) {
+    let state = query.into_inner();
+    match history_info(db_pool.get_ref(), &state) {
         Ok(ls) => HttpResponse::Ok().body(ls),
         Err(_) => HttpResponse::BadRequest().body(""),
     }
 }
```

Comment:
We can use a struct to represent all available query parameters e.g.
```rust
#[derive(Debug, Serialize, Deserialize)]
pub struct HistQueryParams {
    state: String,
}
```
and then use this struct in the function argument
```rust
pub async fn hist(
    query: Query<HistQueryParams>,
    user: Option<User>,
    db_pool: Data<Pool<ConnectionManager<SqliteConnection>>>,
) -> impl Responder {
    info!("Reached handler for /hist route");
    if user.is_none() {
        return HttpResponse::Unauthorized().body("");
    }
    let query = query.into_inner();
    match history_info(db_pool.get_ref(), &query.state) {
        Ok(ls) => HttpResponse::Ok().body(ls),
        Err(_) => HttpResponse::BadRequest().body(""),
    }
}
```